pkg/component/component.go

```diff
@@ -913,40 +914,73 @@ func List(client *occlient.Client, applicationName string, localConfigInfo *conf
 
 // GetComponentFromConfig returns the component on the config if it exists
 func GetComponentFromConfig(localConfig *config.LocalConfigInfo) (Component, error) {
-	if localConfig.ConfigFileExists() {
-		component := getMachineReadableFormat(localConfig.GetName(), localConfig.GetType())
+	component := getComponentFrom(localConfig, localConfig.GetType())
+	if len(component.Name) > 0 {
+		location := localConfig.GetSourceLocation()
+		sourceType := localConfig.GetSourceType()
+		component.Spec.Ports = localConfig.GetPorts()
+		component.Spec.SourceType = string(sourceType)
+		switch sourceType {
+		case config.LOCAL, config.BINARY:
+			component.Spec.Source = util.GenFileURL(location)
+		default:
+			component.Spec.Source = location
+		}
 
-		component.Namespace = localConfig.GetProject()
+		for _, localEnv := range localConfig.GetEnvVars() {
+			component.Spec.Env = append(component.Spec.Env, corev1.EnvVar{Name: localEnv.Name, Value: localEnv.Value})
+		}
 
-		component.Spec = ComponentSpec{
-			App:        localConfig.GetApplication(),
-			Type:       localConfig.GetType(),
-			Source:     localConfig.GetSourceLocation(),
-			Ports:      localConfig.GetPorts(),
-			SourceType: string(localConfig.GetSourceType()),
+		for _, localStorage := range localConfig.GetStorage() {
+			component.Spec.Storage = append(component.Spec.Storage, localStorage.Name)
 		}
+		return component, nil
+	}
+	return Component{}, nil
+}
+
+func GetComponentFromDevfile(info *envinfo.EnvSpecificInfo) Component {
+	if info.Exists() {
+		devfile, err := devfileParser.Parse(info.GetDevfilePath())
+		if err != nil {
+			panic(err)
```

Comment:
Rather than panicking here, can you update `GetComponentFromDevfile` to also return an error value and return an error here?